src/components/button/Button.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+export default function Button({ submit, height, width, color, hover, textColor }) {
+  const [style, setStyle] = useState('');
+  useEffect(() => {
+    setStyle(
+      `text-${textColor} 
+       bg-${color} 
+       hover:bg-${hover} 
+       border  hover:border-${color}  
+       font-bold py-2 px-3 w-${width} 
+       h-${height} 
+       transition duration-500  ease-in-out rounded`
+    );
+  }, []);
+  return (
+    <button type={submit ? 'submit' : 'button'} className={style}>
+      Button
+    </button>
+  );
+}
+Button.propTypes = {
+  submit: PropTypes.bool,
+  height: PropTypes.number,
+  width: PropTypes.number,
+  color: PropTypes.string,
+  textColor: PropTypes.string,
+  hover: PropTypes.string,
+};
+
+Button.defaultProps = {
+  submit: false,
+  height: 30,
+  width: 40,
+  color: 'mainPink',
+  textColor: 'black',
+  hover: 'transparent',
+};
```

Comment:
This is very complicated why are you using useEffect hook !! 
Maybe use something like this. What do you think ?
```
import React from 'react';
const Button = ({ size, varient, type, disabled, text, style }) => {
  const setSize = () => {
    return size === 'lg' ? 'p-4' : 'p-2';
  };
  const setVarient = () => {
    return varient === 'primary' ? 'bg-blue' : 'bg-yellow'; // Or watever you have here.
  };
  return (
    <button type={type} className={`${setSize} ${setVarient}  ${style}`} disabled>
      {text}
    </button>
  );
};
Button.defaultProps = {
  disabled: false,
  size: 'sm',
  type: 'button',
  varient: 'primary',
};
export default Button;
```